snap/validate.go

```diff
@@ -35,3 +35,13 @@ func ValidateName(name string) error {
 	}
 	return nil
 }
+
+// Validate verifies the content in the info.
+func Validate(info *Info) error {
```

Comment:
I think that having a way not to validate (argument on reading yaml? different function) would be very beneficial. This function is used in various places and it's likely that, as we add better validation, more tests will need boilerplate to just have valid-enough data to perform the operation they are interested in.

Plus, I always consider it good if invalid data can be represented. I'd vote for having two top-levels (parse+validate) or separate, explicit, validate step.